R/individual.R

```diff
@@ -283,33 +284,54 @@ optimise_individual_rate_at_size_by_trait <- function(
     # Create an individual object
     types <- extract_RcppR6_template_types(params, "Parameters")
     indv <- do.call("Individual", types)(s) # equiavlent to calling Individual<FF16w,FF16_Env> or FF16_individual(s)
-
-    # set inidividual at specified size
-    indv$set_state(size_name, size)
-    # compute rates given environment
-    indv$compute_rates(env)
     
-    #filter ode rate based on size name
-    res <- indv$ode_rates[indv$ode_names == size_name]
+    if(set_state_by_ode & size_name == "height"){
```

Comment:
should this be `! set_state_by_ode` not the "not" missing

Reply:
My idea here was that in order to be passed through to having height set, rather than grown, you need to both have set_state_by_ode be TRUE AND have the desired property be optimised as height_growth.

Reply:
Makes sense. Except for me, `set_state_by_ode` means grow the plant, not set state directly. Maybe change it to `set_state_directly`? then the logic of the term aligns with your implementation.

Reply:
Ah I see, sorry I misinterpreted the meaning of the expression. Agreed, will change now.

Reply:
Confirming it's a bad name, if we both interpret differently!